[WiFi] Fix properly disconnecting on ESP8266 & fix WiFiAPmode command

TD-er opened the pull request:
Fixes #4118

Apparently the `WifiDisconnect()` function did not perform a true disconnect, causing all kinds of issues.

TD-er committed:
[WiFi] Fix properly disconnecting on ESP8266 * fix WiFiAPmode command

Fixes #4118

tonhuisman commented:
After connecting a client to the AP, I see this message in the log:
```
901508 : Info   : AP Mode: Client connected: 84:98:66:nn:mm:pp Connected devices: 1
902167 : Error  : UNKNOWN WIFI/ETH EVENT: 14
```
I'm sure there is a more useful description available 😉

tonhuisman commented:
And on disconnect:
```
1818203 : Info   : WIFI : Set WiFi to STA
1818205 : Info   : WiFi : AP Stopped
1818309 : Info   : EVENT: WiFi#APmodeDisabled
1818500 : Info   : AP Mode: Client disconnected: 84:98:66:nn:mm:pp Connected devices: 0
```
Maybe the 5 minute AP-mode disable-timer should only be started *after* disconnection of the last client, so there is a chance of reconnection in case of an unintended disconnect? Could also be useful when connecting other ESPEasy instances to this AP.

tonhuisman commented:
AP mode is working smoothly now on both ESP32 and ESP8266 👍

TD-er commented:
> Maybe the 5 minute AP-mode disable-timer should only be started _after_ disconnection of the last client, so there is a chance of reconnection in case of an unintended disconnect? Could also be useful when connecting other ESPEasy instances to this AP.

It does, only it is 60 sec (as far as I can remember).
Or at least it should do this.

tonhuisman commented:
> Or at least it should do this.

The 'AP Stopped' log came *before* the 'Client Disconnected' log, so I don't think it has any delay, as the log lines came immediately when I disconnected.

TD-er commented:
OK, will look at it.
It should reset the "turn AP off" timer to NN sec when it was last used.
Just to make it possible to complete the setup when connected via AP as the connected client may be disconnected when the ESP performs a WiFi scan.

TD-er committed:
[WiFi] Use previously used STA channel when starting AP mode

TD-er committed:
[WiFi] Fix reconnect when toggling AP mode + disable TX power

Disabled any call to set TX power as it seems to seriously do some strange stuff, requiring to reflash the ESP to get it to connect again.

TD-er committed:
Merge branch 'mega' into bugfix/WiFi_AP_mode

TD-er committed:
Merge branch 'mega' into bugfix/WiFi_AP_mode

TD-er committed:
Merge branch 'mega' into bugfix/WiFi_AP_mode

TD-er committed:
Merge branch 'mega' into bugfix/WiFi_AP_mode

TD-er committed:
Merge branch 'mega' into bugfix/WiFi_AP_mode

TD-er committed:
Merge branch 'mega' into bugfix/WiFi_AP_mode

TD-er committed:
Merge branch 'mega' into bugfix/WiFi_AP_mode

TD-er committed:
Merge branch 'mega' into bugfix/WiFi_AP_mode

TD-er commented:
Merging this will probably be incompatible with latest WiFi changes/improvements.
